app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/NationalInsuranceController.scala

```diff
@@ -45,15 +51,29 @@ class NationalInsuranceController @Inject()(
   def showNationalInsuranceNumberForm(): Action[AnyContent] = Action.async { implicit request =>
     withSubscribingAgent { implicit agent =>
       withValidSession { (_, existingSession) =>
-        existingSession.nino match {
-          case Some(nino) =>
-            Ok(html.national_insurance_number(ninoForm.fill(nino)))
-          case None => Ok(html.national_insurance_number(ninoForm))
+        withValidBusinessType(existingSession) {
```

Comment:
Again, this method, called "showNationalInsuranceNumberForm" no longer just does that - can we rename or describe the reasoning behind the new logic?

Reply:
these show methods either just display the page or redirect to some other page  which any show page does, method names are ok I think, if we have to rename the name, we have to rename each and every point because think in terms of auth session
if session doesn't exist we then redirect to GG login , so every endpoint we have to rename  as show...OrRedirectToGGLogin ?

Reply:
Yeah, ok fair enough.  Then please add a description of the reasoning (not the what, but the why), so people understand the reason for the re-routing?  Just a little scaladoc...?